### PR TITLE
Refactor; Put calibration config params.hpp in dr_ensenso.

### PR DESCRIPTION
**Pull request by tijszwinkels**

### CODE REVIEW

File: include/dr_ensenso/calibration_param.hpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	std::string dump_dir;
+};
+
+struct EnsensoCalibrationConfig {
+	InitializeCalibrationConfig initialize_config;
+	bool store_calibration;
+};
+
+DR_PARAM_DEFINE_STRUCT_DECOMPOSITION(EnsensoCalibrationConfig,
+(initialize_config,     "InitializeCalibrationConfig",  "", true)
```
Comment (hgaiser) [inline include/dr_ensenso/calibration_param.hpp]: Indentation.

File: include/dr_ensenso/calibration_param.hpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+(store_calibration,     "bool",                         "", false)
+);
+
+DR_PARAM_DEFINE_STRUCT_DECOMPOSITION(InitializeCalibrationConfig,
+(camera_moving, "bool"  , "", true)
```
Comment (hgaiser) [inline include/dr_ensenso/calibration_param.hpp]: Indentation.

File: include/dr_ensenso/calibration_param.hpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	/// Initial guess of the pattern, can be used to speed up the calibration optimization.
+	/// Depending on the value of camera_moving, this pose is defined in either moving_frame or fixed_frame.
+	std::optional<Eigen::Isometry3d> pattern_guess;
+
+	/// Directory in which calibration information will be stored (leave empty to disable).
+	std::string dump_dir;
+};
+
+struct EnsensoCalibrationConfig {
+	InitializeCalibrationConfig initialize_config;
```
Comment (hgaiser) [inline include/dr_ensenso/calibration_param.hpp]: Missing docstring.

File: include/dr_ensenso/ensenso.hpp [hunk 4]
```diff
@@ -50,6 +50,14 @@ class Ensenso {
 	/// Ensenso calibration result (camera pose, pattern pose, iterations needed, residual error).
 	using CalibrationResult = std::tuple<Eigen::Isometry3d, Eigen::Isometry3d, int, double>;
 
+	/// Camera capture parameters.
+	struct CaptureParams {
+		std::size_t stereo_width;
```
Comment (hgaiser) [inline include/dr_ensenso/ensenso.hpp]: Missing docstring.

File: include/dr_ensenso/ensenso.hpp [hunk 5]
```diff
@@ -192,6 +200,19 @@ class Ensenso {
 	 */
 	cv::Mat loadImage(ImageType type);
 
+	/// Load an image from the camera.
+	/**
+	 * The image must have been captured, retrieved and/or processed before it can be loaded.
+	 */
+	void loadImage(
+		ImageType type,
```
Comment (hgaiser) [inline include/dr_ensenso/ensenso.hpp]: Missing docstring.

File: include/dr_ensenso/ensenso.hpp [hunk 6]
```diff
@@ -192,6 +200,19 @@ class Ensenso {
...
+		std::uint8_t * pointer,
```
Comment (hgaiser) [inline include/dr_ensenso/ensenso.hpp]: Pointer to what? Better to name the variable to what it points to.

File: include/dr_ensenso/calibration_param.hpp [hunk 7]
```diff
@@ -0,0 +1,50 @@
+#pragma once
```
Comment (hgaiser) [inline include/dr_ensenso/calibration_param.hpp]: For now, let's just name the file `param.hpp`. That way we can add other params here if necessary.

File: include/dr_ensenso/ensenso_types.hpp [hunk 8]
```diff
@@ -0,0 +1,59 @@
+#pragma once
```
Comment (hgaiser) [inline include/dr_ensenso/ensenso_types.hpp]: Similarly as `calibration_params.hpp`, let's call this `types.hpp'.

**Comment by hgaiser** [comment]
Oh also, could you add a `CHANGELOG.md` like in other projects to document what has changed?

File: include/dr_ensenso/calibration_param.hpp (hunk 1 above)
Comment (tijszwinkels) [reply include/dr_ensenso/calibration_param.hpp]: ack.

File: include/dr_ensenso/calibration_param.hpp (hunk 3 above)
Comment (tijszwinkels) [reply include/dr_ensenso/calibration_param.hpp]: ack.

File: include/dr_ensenso/ensenso.hpp (hunk 4 above)
Comment (tijszwinkels) [reply include/dr_ensenso/ensenso.hpp]: These were added by @vcarpani and I genuinely don't know the exact context of these parameters. @vcarpani Could you still add these?

File: include/dr_ensenso/ensenso.hpp (hunk 5 above)
Comment (tijszwinkels) [reply include/dr_ensenso/ensenso.hpp]: ack.

File: include/dr_ensenso/ensenso.hpp (hunk 6 above)
Comment (tijszwinkels) [reply include/dr_ensenso/ensenso.hpp]: ack.

File: include/dr_ensenso/calibration_param.hpp (hunk 7 above)
Comment (tijszwinkels) [reply include/dr_ensenso/calibration_param.hpp]: ack.

File: include/dr_ensenso/ensenso_types.hpp (hunk 8 above)
Comment (tijszwinkels) [reply include/dr_ensenso/ensenso_types.hpp]: ack.

File: include/dr_ensenso/ensenso.hpp (hunk 4 above)
Comment (vcarpani) [reply include/dr_ensenso/ensenso.hpp]: alt text

**Comment by hgaiser** [comment]
LGTM!

**Pull request merged**